asciidoctorj-core/src/main/java/org/asciidoctor/extension/Processor.java

```diff
@@ -40,42 +40,59 @@
      * </verbatim>
      * </pre>
      * </p>
+     *
+     * @deprecated Use the annotation {@link ContentModel} instead to describe this Processor
      */
+    @Deprecated
     public static final String CONTENT_MODEL = "content_model";
```

Comment:
In most cases I agree annotations are the preferred approach. It comes in
when constructing a tree manually and building options for the node. (That
could probably be improved too, but a separate issue).